controllers/storagecluster/cephcluster.go

```diff
@@ -800,42 +794,25 @@ func newStorageClassDeviceSets(sc *ocsv1.StorageCluster) []rookCephv1.StorageCla
 			preparePlacement := rookCephv1.Placement{}
 
 			if noPlacement {
-				in := getPlacement(sc, "osd")
-				(&in).DeepCopyInto(&placement)
-
-				if noPreparePlacement {
-					in := getPlacement(sc, "osd-prepare")
-					(&in).DeepCopyInto(&preparePlacement)
-				}
-
-				if len(topologyKeyValues) >= getMinDeviceSetReplica(sc) {
-					// Hard constraints are set in OSD placement for portable volumes with rack failure domain
-					// domain as there is no node affinity in PVs. This restricts the movement of OSDs
-					// between failure domain.
-					if portable && !strings.Contains(topologyKey, "zone") {
-						addStrictFailureDomainTSC(&placement, topologyKey)
```

Comment:
`addStrictFailureDomainTSC()` added the required TSC previously, right? So where was the gap that we are trying to fix? This was ignored if the user set the placement? Does that mean we just need to call this method outside the check for `noPlacement`?

Reply:
No, the strict failure domain just adds another TSC(hard one on the failure domain level) to the osd placement when the failure domain is rack based. Our actual aim is to enforce the TSCs always even when user has specified some placement spec.